src/nativeTest/kotlin/commands/ConnectTest.kt

```diff
@@ -0,0 +1,139 @@
+package commands
+
+import api.IEclairClientBuilder
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ExperimentalCli
+import mocks.DummyEclairClient
+import mocks.DummyResultWriter
+import mocks.FailingEclairClient
+import types.ApiError
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+@OptIn(ExperimentalCli::class)
+class ConnectCommandTest {
+    private fun runTest(eclairClient: IEclairClientBuilder, options: Array<String>): DummyResultWriter {
+        val resultWriter = DummyResultWriter()
+        val command = ConnectCommand(resultWriter, eclairClient)
+        val parser = ArgParser("test")
+        parser.subcommands(command)
+        parser.parse(options)
+        return resultWriter
+    }
+
+    @Test
+    fun `successful request via URI`() {
+        val resultWriter = runTest(
+            DummyEclairClient(),
+            arrayOf(
+                "connect",
+                "--host",
+                "http://localhost:8080",
+                "-p",
+                "password",
+                "--uri",
+                "02f666711319435b7905dd77d10c269d8d50c02668b975f526577167d370b50a3e@127.0.0.1:9736"
+            )
+        )
+        assertNull(resultWriter.lastError)
+        assertEquals(DummyEclairClient.validConnectResponse, resultWriter.lastResult)
+    }
+
+    @Test
+    fun `successful request via NodeId`() {
+        val resultWriter = runTest(
+            DummyEclairClient(),
+            arrayOf(
+                "connect",
+                "--host",
+                "http://localhost:8080",
+                "-p",
+                "password",
+                "--nodeId",
+                "02f666711319435b7905dd77d10c269d8d50c02668b975f526577167d370b50a3e"
+            )
+        )
+        assertNull(resultWriter.lastError)
+        assertEquals(DummyEclairClient.validConnectResponse, resultWriter.lastResult)
+    }
+
+    @Test
+    fun `successful request via Manual`() {
+        val resultWriter = runTest(
+            DummyEclairClient(),
+            arrayOf(
+                "connect",
+                "--host",
+                "http://localhost:8080",
+                "-p",
+                "password",
+                "--nodeId",
+                "02f666711319435b7905dd77d10c269d8d50c02668b975f526577167d370b50a3e",
+                "--address",
+                "127.0.0.1"
```

Comment:
I think I am not calling the right address, and hence I'm getting the `address not found` message when I try using the `connect manually`. Can you please provide some details on it?